numba/parfors/parfor.py

```diff
@@ -3969,7 +3970,14 @@ def get_user_varname(v):
 
     # TODO: make sure parfor1's reduction output is not used in parfor2
     # only data parallel loops
-    if has_cross_iter_dep(parfor1) or has_cross_iter_dep(parfor2):
+    func_ir._definitions = build_definitions(func_ir.blocks)
```

Comment:
I guess in this case it is after a parfor is fused.  So, I moved this to after fusion is done but I wonder if there are other cases where invalidating changes are made and this isn't updated.

Reply:
Turns out my attempt didn't work.  I put it back the way it was which works.  It would be nice to easily know what invalidates the previous version and only rebuild defs right after that but that isn't necessarily time preserving if only a few places need the defs.  Maybe we need a better infrastructure for invalidating the defs on modification and then caching the defs until modification and if you ask for the defs after invalidation then rebuild.  I'd like to get this in now though.